### ISSUE TITLE
A type error is being reported even though "type checking mode" is set to "off"

### ISSUE THREAD

**Issue by theroggy** [microsoft/pyright#8453]
## Environment data

-   Pylance version: v2024.7.1
-   OS and version: Windows 10 (10.0.19045 Build 19045)
-   Python version (& distribution if applicable, e.g. Anaconda): python 3.11.9, h631f459_0_cpython, conda-forge

## Code Snippet

``` py
from numpy.typing import NDArray
from shapely.geometry.base import BaseGeometry


def simplify_topo(test: NDArray[BaseGeometry]):
    return

```

## Repro Steps

`NDArray[BaseGeometry]` in the above snippet will be highlighted as having an error.

## Expected behavior

No type errors, as "type checking mode" is set to "off".

## Actual behavior

This type error is shown:

```
Could not specialize type "NDArray[_ScalarType_co@NDArray]"
  Type "BaseGeometry" is incompatible with type "generic"
    "BaseGeometry" is incompatible with "generic"
```

## Logs

No significant logging is being written...

**Comment by erictraut** [comment]
I did a full pass over all diagnostics emitted by the type evaluator and made sure that every one had an associated diagnostic rule. This allows the severity of all type evaluator diagnostics to be under user control. This will be addressed in the next release of pyright.

**Comment by Cnly** [comment]
Hmmm, I wonder if there's a regression? I'm using coc-pyright and I think I'm seeing this on 1.1.37{6,7,8}. Rolling pyright back to 1.1.375 solved the problem.

**Comment by Cnly** [comment]
@erictraut Thank you for the rapid reply and detailed explanation! I indeed have a `pyrightconfig.json` in my project. Adding `"typeCheckingMode": "off"` to the file fixed the problem on 1.1.378.
